lib/assessment.js

```diff
@@ -8,33 +8,87 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
+var min = function min() {previousValue, currentValue, index, array)
+return previousValue < currentValue ? previousValue : currentValue;
+};
+
 
+// callback(previousValue, currentValue, index, array)
+var reduce = function reduce(array, callback, initialValue) {
+  var previousValue;
+  if (initialValue !== undefined) {
+    previousValue = initialValue;
+  } else {
+    array = array.slice();
+    previousValue = array.shift();
+  }
 };
 
+console.log(reduce(numbers, min));
+
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
+var sum = function sum(previousValue, currentValue, index, array) {
+  return previousValue + currentValue;
+};
+
+var reduce = function reduce(array, callback, initialValue) {
+  var previousValue;
+  if (initialValue !== undefined) {
+    previousValue = initialValue;
+  } else {
+    array = array.slice();
+    previousValue = array.shift();
+  }
+
+  array.forEach(function(currentValue, index, array) {
+    previousValue = callback(previousValue, currentValue, index, array);
+  });
 
+  return previousValue;
 };
 
+var numbers = [1, 2, 3, 4, 5];
+
+console.log(reduce(numbers, sum));
+);
+
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 var any = function any(array, callback, thisArg) {
+  var result = true;
+  array.forEach(function(currentValue, index, array) {
+    // if callback ever returns false, result will be set
+    // to false amd the callback will not be called in any
+    // subsequent iterations.
+    result = result || callback(currentValue, index, array);
```

Comment:
Great, you noted that the principal difference between `any` and `every` is that any should be ORing the boolean results into an accumulator variable, not ANDing them. However, `result`'s initial value being `true` breaks this, because `true` OR anything yields `true`. So, if you change line 62 so that it initializes `result` to `false`, your `any` function will be entirely correct. 9/10 credit.